crates/holochain/src/core/sys_validate/tests.rs

```diff
@@ -610,30 +612,48 @@ async fn test_agent_update() {
     let head = chain.chain_head().unwrap().unwrap();
 
     let a1 = Action::AgentValidationPkg(AgentValidationPkg {
-        author: agent2.clone(),
+        author: agents[0].clone(),
         timestamp: (head.timestamp + sec).unwrap(),
         action_seq: head.seq + 1,
         prev_action: head.action,
         membrane_proof: None,
     });
 
     let a2 = Action::Create(Create {
-        author: agent2.clone(),
+        author: agents[0].clone(),
         timestamp: (a1.timestamp() + sec).unwrap(),
         action_seq: a1.action_seq() + 1,
         prev_action: a1.to_hash(),
         entry_type: EntryType::AgentPubKey,
-        entry_hash: agent2.clone().into(),
+        entry_hash: agents[1].clone().into(),
         weight: EntryRateWeight::default(),
     });
 
-    let a3 = Action::Create(Create {
-        author: agent3.clone(),
-        timestamp: (a2.timestamp() + sec).unwrap(),
+    let a3 = Action::AgentValidationPkg(AgentValidationPkg {
+        author: agents[1].clone(),
+        timestamp: (head.timestamp + sec).unwrap(),
         action_seq: a2.action_seq() + 1,
         prev_action: a2.to_hash(),
+        membrane_proof: None,
+    });
+
+    let a4 = Action::Create(Create {
+        author: agents[1].clone(),
+        timestamp: (a1.timestamp() + sec).unwrap(),
+        action_seq: a3.action_seq() + 1,
+        prev_action: a3.to_hash(),
         entry_type: EntryType::AgentPubKey,
-        entry_hash: agent3.clone().into(),
+        entry_hash: agents[2].clone().into(),
+        weight: EntryRateWeight::default(),
+    });
+
+    let a5 = Action::Create(Create {
+        author: agents[2].clone(),
```

Comment:
agent[2] attempts to create a new agent without having been validated, hence it should fail